lookout/core/metrics.py

```diff
@@ -1,12 +1,197 @@
+import os
+import string
+from threading import Lock
 from typing import Union
 
+from prometheus_client import start_http_server
+from prometheus_client.metrics import MetricWrapperBase
 
-def submit_event(key: str, value: Union[int, float, bool]):
+
+PROMETHEUS_SERVER = None
+
+PROMETHEUS_HOST = os.getenv("PROMETHEUS_HOST", "0.0.0.0")
+
+PROMETHEUS_PORT = int(os.getenv("PROMETHEUS_PORT", "8000"))
+
+
+class MutexValue:
+    """A float protected by a mutex."""
+
+    _multiprocess = False
+
+    def __init__(self, typ, metric_name, name, labelnames, labelvalues, **kwargs):
+        """Store values protected by a lock."""
+        self._value = 0.0
+        self._lock = Lock()
+        self._interm_sum = 0
+
+    def add(self, amount):
+        """Increase the value by the given amount."""
+        with self._lock:
+            summ = self._value
+            y = amount - self._interm_sum
+            t = summ + y
+            self._interm_sum = (t - summ) - y
+            self._value = t
+
+    def set(self, value):
+        """Set the value to a given amount."""
+        with self._lock:
+            self._interm_sum = 0
+            self._value = value
+
+    def get(self):
+        """Read the internal value."""
+        with self._lock:
+            return self._value
+
+
+class ConfidentCounter(MetricWrapperBase):
+    """It stores the count, cumulative sum, and the cumulative sum of squares of a variable.
+
+    It makes it possible to calculate the rolling mean and standard deviation
+    of a time series.
+    """
+
+    _type = "counter"
+
+    def _metric_init(self):
+        self._count = MutexValue(
+            self._type, self._name, self._name + "_count", self._labelnames, self._labelvalues,
+        )
+        self._sum = MutexValue(
+            self._type, self._name, self._name + "_sum", self._labelnames, self._labelvalues,
+        )
+
+        self._square = MutexValue(
+            self._type,
+            self._name,
+            self._name + "_sum_of_squares",
+            self._labelnames,
+            self._labelvalues,
+        )
+
+    def add(self, amount):
+        """Keep track of the given value."""
+        self._count.add(1)
+        self._sum.add(amount)
+        self._square.add(amount ** 2)
+
+    def _child_samples(self):
+        count = max(self._count.get(), 1)
+        _sum = self._sum.get()
+        square = self._square.get()
+        return ("_count", {}, count), ("_sum", {}, _sum), ("_sum_of_squares", {}, square)
+
+
+def start_prometheus(port: int = PROMETHEUS_PORT, addr: str = PROMETHEUS_HOST):
```

Comment:
```suggestion
def start_prometheus(host: str, port: int):
```
1. `addr` -> `host`
2. The host always comes first
3. No defaults